integtest/spec/helper/dsl/convert_single.rb

```diff
@@ -9,21 +9,19 @@ module ConvertSingle
     # file to convert. It does the conversion with both with `--asciidoctor`
     # and without `--asciidoctor` and asserts that the files are the same.
     def convert_single_before_context
-      include_context 'source and dest'
-      before(:context) do
-        from = yield @src.repo('src')
-        @src.init_repos
-        @asciidoctor_out = @dest.convert_single from, '.', asciidoctor: true
+      convert_before do |src, dest|
+        from = yield src.repo('src')
+        src.init_repos
+        dest.convert_single from, '.', asciidoctor: true
         # Convert a second time with the legacy `AsciiDoc` tool and stick the
         # result into the `asciidoc` directory. We will compare the results of
         # this conversion with the results of the `Asciidoctor` conversion.
-        @asciidoc_out = @dest.convert_single from, 'asciidoc',
-                                             asciidoctor: false
+        dest.convert_single from, 'asciidoc', asciidoctor: false
       end
       include_examples 'convert single'
     end
     shared_context 'convert single' do
-      let(:out) { @asciidoctor_out }
+      let(:out) { outputs[0] }
```

Comment:
Most users of this context won't know or care that there are is more than one output.